### PR TITLE
Fix open with feature

### LINKED ISSUE #1008: As an user I want to be able to "open with" files from OC to third party apps and working with MDM so I can keep sharing with mail system app and also support whitelisted apps
AC:

- At least with iOS11 be able to share only with white listed apps (Enhancement)
- With all versions without white listed apps enabled be able to share with all system apps as mail app, save image.. (Regression within 3.7 after add previous enhancement in versions <iOS11, related https://github.com/owncloud/ios/issues/980)

Dietmar S., thank you for your feedback

Regression of 3.6.2 within 3.7.0

TASKS:
- [X] [DEV] Create branch  fix_openWith_feature
- [X] [DEV] Fix 'open with', show more options as mail app, in versions <iOS11
- [X] [DEV] Fix 'whitelisted' apps available also in versions >iOS11
- [X] [DEV] Fix 'open with' menu sometimes not appears
- [X] [QA] Perform test plan: QA repo: https://github.com/owncloud/QA/blob/master/Mobile/iOS/Release_3.7.3/1010-open_in_wrapped.md @jesmrec 
- [ ] [DEV] Merge branch  fix_openWith_feature

PR: https://github.com/owncloud/ios/pull/1010

### PR DESCRIPTION
**Pull request by nasli**
Fix #1008 
- [x] Fix, all menu options appears in all iOS versions (as mail app, save image,..)  (select share to mail, appears, close mail view, share again)
- [x] Fix, only appears whiteListed apps selected with MDM

### CODE REVIEW

**Comment by jesmrec** [comment]
Checks to perform:

| | iOS10 | iOS11 | 
| :-: | :---- | :-------- | 
|**wrapped**| iPhone:  <br>iPad: |iPhone:  <br>iPad:|
| **non-wrapped** |iPhone:  <br>iPad:|iPhone:  <br>iPad:|

**Comment by jesmrec** [comment]
Test plan: https://github.com/owncloud/QA/blob/master/Mobile/iOS/Release_3.7.3/1010-open_in_wrapped.md

**Comment by jesmrec** [comment]
Last checks performed with iPad - iOS10 and wrapped app, all ok.

Approved

File: Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m [hunk 1]
```diff
@@ -845,11 +845,7 @@ - (void)presentWhiteView {
     
     _titleLabel.text = @"";
 
-    if (k_use_open_with_UIDocumentInteractionController) {
-        [_openWith.documentInteractionController dismissMenuAnimated:YES];
-    } else {
-        [_openWith.activityPopoverController dismissPopoverAnimated:YES];
-    }
+    [_openWith.documentInteractionController dismissMenuAnimated:YES];
```
Comment (davigonz) [inline Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m]: You could check here if _openWith object exists, as you do below.

File: Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m [hunk 2]
```diff
@@ -1747,11 +1736,7 @@ - (void) customWillRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterf
     
     if (_openWith) {
         
-        if (k_use_open_with_UIDocumentInteractionController) {
-            [_openWith.documentInteractionController dismissMenuAnimated:NO];
-        } else {
-            [_openWith.activityPopoverController dismissPopoverAnimated:NO];
-        }
+        [_openWith.documentInteractionController dismissMenuAnimated:NO];
```
Comment (davigonz) [inline Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m]: You can check here not only that **_openWith** exists but also that **_openWith.documentInteractionController** exists.

File: Owncloud iOs Client/Tabs/FileTab/FilesViewController.m [hunk 3]
```diff
@@ -624,11 +624,7 @@ - (BOOL)shouldAutorotate {
 - (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration{
     
     if (self.openWith && !IS_IPHONE) {
-        if (k_use_open_with_UIDocumentInteractionController) {
-            [self.openWith.documentInteractionController dismissMenuAnimated:NO];
-        } else {
-            [self.openWith.activityView dismissViewControllerAnimated:NO completion:nil];
-        }
+        [self.openWith.documentInteractionController dismissMenuAnimated:NO];
```
Comment (davigonz) [inline Owncloud iOs Client/Tabs/FileTab/FilesViewController.m]: Check if **openWith.documentInteractionController** exists?

File: Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m (hunk 1 above)
Comment (nasli) [reply Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m]: 👍

File: Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m (hunk 2 above)
Comment (nasli) [reply Owncloud iOs Client/Files/Preview/DetailView/DetailViewController.m]: 👍

File: Owncloud iOs Client/Tabs/FileTab/FilesViewController.m (hunk 3 above)
Comment (nasli) [reply Owncloud iOs Client/Tabs/FileTab/FilesViewController.m]: 👍

**Comment by nasli** [comment]
Thanks @davigonz! , please review it again with the changes

**Comment by davigonz** [comment]
LGTM

**Pull request merged**